src/idl_gen_general.cpp

```diff
@@ -203,8 +203,13 @@ class GeneralGenerator : public BaseGenerator {
       const std::string &classcode, bool needs_includes) {
     if (!classcode.length()) return true;
 
-    std::string code;
-    code = code + "// " + FlatBuffersGeneratedWarning();
+    std::string code("//------------------------------------------------------------------------------\n"
```

Comment:
See [here](https://github.com/dotnet/roslyn/blob/d2beae6970b65797bfbeac4f82334fd1f3e89c13/src/Compilers/Core/Portable/InternalUtilities/GeneratedCodeUtilities.cs#L13) and [here](https://github.com/dotnet/roslyn/blob/d2beae6970b65797bfbeac4f82334fd1f3e89c13/src/Compilers/Core/Portable/InternalUtilities/GeneratedCodeUtilities.cs#L87-L94) for the compiler implementation, if you are curious.